python/stack_plots.py

```diff
@@ -164,29 +182,49 @@ def draw_plot(plotname="fatjet_msoftdrop", title="myTitle", log=True, compare_da
     TTXplot.Add(TTHbbplot)
 
     totalSM = ZToMuMuplot.Clone("totalSM")
+    totalSM.Add(ttbarplot)
     totalSM.Add(ST_tWplot)
     totalSM.Add(TTXplot)
-    totalSM.Add(ttbarplot)
-    totalSM.Add(ZZplot)
     totalSM.Add(WWplot)
+    totalSM.Add(ZZplot)
     totalSM.Add(WZplot)
     totalScale   = totalSM.Integral(0,-1)
     if "cutflow" in plotname:
         totalScale = totalSM.GetBinContent(1)
 
     if args.shape and totalScale>0.0:
         ZToMuMuplot.Scale(1.0/totalScale)
+        ttbarplot.Scale(1.0/totalScale)
         ST_tWplot.Scale(1.0/totalScale)
         TTXplot.Scale(1.0/totalScale)
-        ttbarplot.Scale(1.0/totalScale)
-        ZZplot.Scale(1.0/totalScale)
         WWplot.Scale(1.0/totalScale)
+        ZZplot.Scale(1.0/totalScale)
         WZplot.Scale(1.0/totalScale)
 
+    signalXSecScale = { }
+    if log==False and args.signalScale and not args.shape:
+        for i,mass in enumerate(args.signalMass):
```

Comment:
What are trying to optimize with this logic? Compute the signal scale, so that thee signal doesn't overshoot over the SM but isn't too low?

Reply:
Yes, I came up with this by trial, trying to avoid signal being too low as well as being too high compared to SM, so that both are better visible in the plot.